### ISSUE TITLE
EINTR while reading process list

### ISSUE THREAD

**Issue by walles** [walles/px#89]
I'm guessing this is the `SIGWINCH` that's interrupting the read, it has happened a number of times during window resizing.

```
Problems detected, please send this text to one of:
* https://github.com/walles/px/issues/new
* johan.walles@gmail.com

px version: 1.5.0-7-g6671454

Python version: 2.7.16 (default, Jun 18 2021, 03:23:53)
[GCC Apple LLVM 12.0.5 (clang-1205.0.19.59.6) [+internal-os, ptrauth-isa=deploy

Platform info: Darwin-20.6.0-x86_64-i386-64bit

2021-08-15 07:21:52CEST ERROR    Running ptop failed
Traceback (most recent call last):
  File "./px.pex/px/px_top.py", line 482, in top
    _top(search=search)
  File "./px.pex/px/px_top.py", line 469, in _top
    current = px_process.get_all()
  File "./px.pex/px/px_process.py", line 454, in get_all
    for ps_line in stdout:
IOError: [Errno 4] Interrupted system call⏎
```

**Comment by walles** [comment]
Can repro with 1.4.0 and 1.5.0 as well, applying this patch and resizing the window vigorously should repro the problem:

```diff
diff --git px/px_top.py px/px_top.py
index 47f1c55..9fd364b 100644
--- px/px_top.py
+++ px/px_top.py
@@ -448,7 +448,7 @@ def _top(search=""):
         toplist = get_toplist(baseline, current, sort_by_memory)
         redraw(toplist, launchcounter, rows, columns)

-        command = get_command(timeout_seconds=1)
+        command = get_command(timeout_seconds=0)

         # Handle all keypresses before refreshing the display
         while command is not None:
@@ -465,7 +465,7 @@ def _top(search=""):
         global last_process_poll
         now = time.time()
         delta_seconds = now - last_process_poll
-        if delta_seconds >= 0.8:
+        if delta_seconds >= 0:
             current = px_process.get_all()
             last_process_poll = now
```

**Comment by walles** [comment]
Unable to repro any more, possibly fixed as part of the performance work in [1.5.2](https://github.com/walles/px/releases/tag/1.5.2), which reworked window resize handling.
